TTInAppPurchases/View Controllers/SubscriptionFlow/WeeklyMonthlyAndAnnualViewController.swift

```diff
@@ -0,0 +1,318 @@
+//
+//  WeeklyMonthlyAndAnnualViewController.swift
+//  TTInAppPurchases
+//
+//  Created by Revathi on 18/04/22.
+//
+
+import UIKit
+import Lottie
+import NVActivityIndicatorView
+import CoreAudio
+
+public class WeeklyMonthlyAndAnnualViewController: UIViewController, SubscriptionViewControllerProtocol {
+      
+    //MARK: - IBOutlets
+    @IBOutlet weak var animationView: UIView!
+    @IBOutlet weak var contentView: UIView!
+    @IBOutlet weak var cancelButton: UIButton!
+    @IBOutlet weak var subscribeButton: UIButton!
+    @IBOutlet weak var primaryHeaderLabel: UILabel!
+    @IBOutlet weak var feature1Label: UILabel!
+    @IBOutlet weak var feature2Label: UILabel!
+    @IBOutlet weak var feature3Label: UILabel!
+    @IBOutlet weak var feature4Label: UILabel!
+    @IBOutlet weak var stackViewHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var freeTrialLabel: UILabel!
+    @IBOutlet weak var firstSubscriptionButton: UIButton!
+    @IBOutlet weak var secondSubscriptionButton: UIButton!
+    @IBOutlet weak var thirdSubscriptionButton: UIButton!
+    @IBOutlet weak var topDescriptionLabel: UILabel!
+    @IBOutlet weak var bottomHeaderLabel: UILabel!
+    @IBOutlet weak var bottomDescriptionLabel: UILabel!
+    @IBOutlet var priceButtons: [UIButton]!
+    @IBOutlet var tickMarkImageViews: [UIImageView]!
+    
+    //MARK: IBOutlet Collections
+    @IBOutlet var priceButtonsZoomedHeight: [NSLayoutConstraint]!
+    @IBOutlet var priceButtonStandardHeight: [NSLayoutConstraint]!
+    
+    //MARK: External Parameters
+    public weak var delegate: SubscriptionViewControllerDelegate?
+    public weak var uiProviderDelegate: UpgradeUIProviderDelegate?
+    public weak var specialOfferUIProviderDelegate: SpecialOfferUIProviderDelegate?
+    public var giftOffer: Bool = false
+    public var hideCloseButton: Bool = false
+    
+    //MARK: Internal Parameters
+    private let lottieView = AnimationView(name: "HelloAnimation")
+    private let bounds = UIScreen.main.bounds
+    private var featureLabelTextStyle: UIFont.TextStyle = .callout
+    
+    private var _selectedIndex = 1 {
+        didSet {
+            if isViewLoaded {
+                
+                checkFreeOfferTrialStatus(for: _selectedIndex)
+                
+                switch(_selectedIndex) {
+                case 0:
```

Comment:
You can simplify this

`unhighlightButton(at: 0)
unhighlightButton(at: 1)
unhighlightButton(at: 2)
highlightButton(at: selectedIndex)`

you won't need a switch case in this case.